app/ceom/maps/templates/maps/pixels_validation.html

```diff
@@ -0,0 +1,158 @@
+        }
+    }    
+</style>
+<nav>
+    <h2 class="legend title">Pixels Validation</h2>
+    <p style="font-size:large">Here you can retrieve geographical data of pixels in the selected area. The file of data you get is in *.csv file</p>
+    <div class="my-2">
+        <p><strong style="font-size:large"> 1. Select size of region. </strong></p>
+        <ul style="list-style: none;">
+            <li>
+                <p><input type="radio" name="boxSize" value="5" checked> 5 degrees</p>
+            </li>
+            <li>
+                <p><input type="radio" name="boxSize" value="10"> 10 degrees</p>
+            </li>
+        </ul>
+        <p><strong style="font-size:large">2. Zoom-in the map and then click on a point of interest </strong></p>
+        <p>
+            <div id="map"></div>
+            <div class="row">
+                <div class="col-sm-3">
+                    <input type="checkbox" id="auto_zoom" name="auto_zoom" checked /> Zoom to selected region</div>
+            </div>
+        </p> 
+        <p><strong style="font-size:large">3. Check your request and click Download</strong></p>
+        <div>
+            <button class="btn btn-success mb-2" onclick="pixels_validation_csv()"><i class="icon-plus-sign "></i>Download CSV</button>
+        </div>
+    </div>
+</nav>
+<script>
+    let map;
+    let polygon;
+    let lat = 0;
+    let lon = 0;
+    let centerLocation = {
+        lat: lat,
+        lng: lon
+    };
+    var size = 5;
+	function initMap() 
+	{
+        map = new google.maps.Map(document.getElementById('map'), {
+          zoom: 3,
+          center: centerLocation,
+          gestureHandling: 'cooperative',
+          mapTypeId: 'satellite',
+        });
+    
+        marker = new google.maps.Marker({
+            position: centerLocation,
+            map: map,
+        });
+        
+        polygon = new google.maps.Polygon({ 
+            strokeColor: "#000000",                 
+            strokeOpacity: 0.9,
+            strokeWeight: 2,                                 
+            fillOpacity: 0,
+            clickable: false,                 
+        });
+        updatePolygon(lat, lon);
+        google.maps.event.addListener(map, 'click', async function (evt) {
+            lat = evt.latLng.lat();
+            lon = evt.latLng.lng();
+            marker.setPosition({lat: lat, lng: lon});
+            if($("#auto_zoom").is(":checked")) {
+                let mzs = new google.maps.MaxZoomService();
+                let zoomobj = await mzs.getMaxZoomAtLatLng(marker.getPosition());
+                map.setZoom(Math.min(zoomobj.zoom, 5));
+                map.setCenter(marker.getPosition());     
+            }
+            updatePolygon(lat, lon);
+        });    
+        marker.setMap(map);
+        polygon.setMap(map);
+    };
+
+    $(document).on('input', function(e) {
+        centerLocation = marker.getPosition();
+        size = $('input[name=boxSize]:checked').val();
```

Comment:
Very good use of JQuery here, good work

I'm not sure about using initMap() in this listener though, you are completely rebuilding the map when the user changes the polygon size. This seems like overkill to me. 

You can imagine a user is trying to select an area in the USA, then decides they need a bigger box, so they click on your radio button, which resets them back to (0,0), and now they have to find the area again

You can replace initMap() here with some code that just modifies the existing polygon on the map
It might be tricky but it is possible

Reply:
It doesn't reset back to (0,0). I assigned centerLocation = present location and in the initMap function, I use centerLocation to assign the center of map and the position of marker . Let me try the other way to see if I can make it better

Reply:
I fixed it. I was overthinking so I made things complicated